src/kostentraeger/index.ts

```diff
@@ -0,0 +1,410 @@
+
+type InstitutionListWithValidityStartDate = {
+    validityStartDate: Date,
+    institutions: Institution[]
+}
+
+/** An index to find Kostenträger information for Pflegedienstleister and sonstige Dienstleister */
+export class KostentraegerIndex {
+
+    /* The lists are deliberately not merged together because it is neither invalid nor unrealistic
+       that one and the same institution is mentioned in different Kostenträger files, each with 
+       different links and maybe other data too.
+       For example, the companies Telekom and Bitmarck are Datenannahmestellen (=data 
+       acceptance offices) for multiple health care insurances.
+     */
+
+    private index = new Map<LeistungserbringergruppeSchluessel,
+                            Map<KassenartSchluessel, InstitutionListWithValidityStartDate[]>
+                           >()
+
+    constructor(institutionLists: InstitutionList[]) {
+        institutionLists.forEach((institutionList) => {
+            const leGruppe = institutionList.leistungserbringerGruppeSchluessel
+            const kassenart = institutionList.kassenart
+            // lazily create those maps...
+            if (!this.index.has(leGruppe)) {
+                this.index.set(leGruppe, new Map<KassenartSchluessel, InstitutionListWithValidityStartDate[]>())
+            }
+            const institutionListForLeGruppe = this.index.get(leGruppe)!
+            if (!institutionListForLeGruppe.has(kassenart)) {
+                institutionListForLeGruppe.set(kassenart, [])
+            }
+            const institutionListForKassenart = institutionListForLeGruppe.get(kassenart)!
+            institutionListForKassenart.push({
+                validityStartDate: institutionList.validityStartDate,
+                institutions: institutionList.institutions,
+            })
+        })
+    }
+
+    /** Find information on the Kostenträger for sending the given data type
+     * 
+     * @param dataType What shall be sent: digital receipts, paper receipts, prescriptions,... etc
+     *        0 for digital receipts.
+     * 
+     * @param pflegekasseIK 9-digit "Institutionskennzeichen" of the care insurance of the insuree
+     * 
+     * @param leistungsart Type of health care service provided. Either a health care service type 
+     *        (Leistungsart) from SGB XI or a from SGB V (Abrechnungscode).
+     * 
+     * @param location Location of the health care service provider
+     * 
+     * @param date Date at which the receipt should be sent. Optional, defaults to now. Some
+     *        institutions given in the Kostenträger files have a validity date range.
+     */
+    find(
+        paperDataType: PaperDataType,
+        pflegekasseIK: string,
+        leistungsart: Leistungsart,
+        location: CareProviderLocationSchluessel,
+        date: Date = new Date()
+
+    ): KostentraegerFindResult | undefined {
```

Comment:
This is the streamlined interface to query the data in the end, @michaelkamphausen